http-gateway/pkg/server/topic_existence_checker.go

```diff
@@ -33,29 +43,67 @@ func NewAlwaysTrueTopicExistenceChecker() TopicExistenceChecker {
 // NewRiffTopicExistenceChecker configures a TopicExistenceChecker using the
 // provided Clientset.
 func NewRiffTopicExistenceChecker(clientSet *versioned.Clientset) TopicExistenceChecker {
-	return &riffTopicExistenceChecker{client: clientSet}
-}
+	mutex := &sync.Mutex{}
+
+	riffInformerFactory := informers.NewSharedInformerFactory(clientSet, time.Second*30)
+	topicInformer := riffInformerFactory.Projectriff().V1alpha1().Topics()
+
+	knownTopics := make(map[string]ignoredValue)
 
-func (tec *alwaysTrueTopicExistenceChecker) TopicExists(namespace string, topicName string) (bool, error) {
-	return true, nil
+	topicInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			mutex.Lock()
+			defer mutex.Unlock()
+
+			//TODO: implement riff-specific KeyFunc https://github.com/projectriff/riff/pull/558#discussion_r184437224
+			key, err := cache.MetaNamespaceKeyFunc(obj)
+			if err != nil {
+				// It is likely that the key is faulty, but we cannot signal an error.
+				// To prevent errors for processing a bad key, we return after logging.
+				log.Printf("AddFunc failed during a topic lookup: %#v", err)
+				return
+			}
+
+			knownTopics[key] = ignoredValue{}
+			log.Printf("New topic has been added: %s", key)
+		},
+
+		DeleteFunc: func(obj interface{}) {
+			mutex.Lock()
+			defer mutex.Unlock()
+
+			//TODO: implement riff-specific KeyFunc https://github.com/projectriff/riff/pull/558#discussion_r184437224
+			key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+			if err != nil {
+				// It is likely that the key is faulty, but we cannot signal an error.
+				// To prevent errors for processing a bad key, we return after logging.
+				log.Printf("DeleteFunc failed during a topic lookup: %#v", err)
+				return
+			}
+
+			delete(knownTopics, key)
+			log.Printf("A topic was removed: %s", key)
+		},
+	})
+
+	done := make(chan struct{}) //TODO: ideally, this would be the same channel used by the gateway itself
```

Comment:
We don't seem to close `done` anywhere.

Reply:
Time to act on my TODO. I had forgotten to come back and work out if there was a way to do it cleanly without widening the `NewRiffTopicExistenceChecker` signature.